annotations/annotations_rw.go

```diff
@@ -66,7 +68,49 @@ func (rw *annotationsRW) Read(ctx context.Context, contentUUID string) ([]*Annot
 	case http.StatusNotFound:
 		return nil, false, nil
 	default:
-		return nil, false, fmt.Errorf("annotations RW returned an unexpected HTTP status code: %v", resp.StatusCode)
+		return nil, false, fmt.Errorf("annotations RW returned an unexpected HTTP status code in read operation: %v", resp.StatusCode)
+	}
+}
+
+func (rw *annotationsRW) Write(ctx context.Context, contentUUID string, annotations []Annotation) error {
+	tid, err := tidUtils.GetTransactionIDFromContext(ctx)
+
+	if err != nil {
+		tid = tidUtils.NewTransactionID()
+		log.WithField(tidUtils.TransactionIDKey, tid).
+			WithField("uuid", contentUUID).
+			WithError(err).
+			Warn("Transaction ID error in writing annotations to RW with concept data: Generated a new transaction ID")
+		ctx = tidUtils.TransactionAwareContext(ctx, tid)
+	}
+
+	writeLog := log.WithField(tidUtils.TransactionIDKey, tid).WithField("uuid", contentUUID)
+
+	annotationsBody, err := json.Marshal(annotations)
+	if err != nil {
+		writeLog.WithError(err).Error("Unable to marshall annotations that needs to be written")
+		return err
+	}
+
+	req, err := http.NewRequest("PUT", rw.endpoint+"/drafts/content/"+contentUUID+"/annotations", bytes.NewBuffer(annotationsBody))
```

Comment:
Should the path part of the write URL be hard-coded here? Maybe it should be combined into the command-line parameter which would then be a URL template. (I realise this code mirrors the read - so in fact the hard-coding is now in two places!)

Reply:
I am happy to use `fmt.Sprintf("%s/drafts/content/%s/annotations",rw.endpoint,contentUUID)`, but it looks like that it is quite less efficient that concatenating strings with `+`.

Reply:
Leave it as is.

Reply:
Fixed. Specified the string as a pattern in a constant